### PR TITLE
Merkle Tree Pallet Benchmark

### PR DESCRIPTION
**Pull request by Wizdave97**

### CODE REVIEW

File: pallets/mt/src/lib.rs [hunk 1]
```diff
@@ -277,24 +293,29 @@ pub mod pallet {
...
 
-		#[pallet::weight(0)]
+		#[pallet::weight(T::WeightInfo::force_set_default_hashes(default_hashes.len() as u32))]
 		pub fn force_set_default_hashes(
 			origin: OriginFor<T>,
 			default_hashes: Vec<T::Element>,
 		) -> DispatchResultWithPostInfo {
 			T::ForceOrigin::ensure_origin(origin)?;
-			// set the new maintainer
+			let len_of_hashes = default_hashes.len();
```
Comment (shekohex) [inline pallets/mt/src/lib.rs:313]: You should ensure here, also, that the `len_of_hashes == T::MaxTreeDepth::get()`.

File: pallets/mt/src/tests.rs [hunk 2]
```diff
@@ -37,6 +38,22 @@ fn should_fail_in_case_of_larger_depth() {
...
+fn should_fail_in_case_when_max_default_hashes_is_exceeded() {
+	new_test_ext().execute_with(|| {
+		let max_default_hashes = <Test as Config>::MaxDefaultHashes::get();
+		assert_err!(
+			MerkleTree::force_set_default_hashes(Origin::root(), vec![<Test as Config>::DefaultZeroElement::get(); (max_default_hashes + 1) as usize]),
+			DispatchError::Module {
+				index: 3,
+				error: 5, // ExceedsMaxDefaultHashes,
+				message: None,
+			}
```
Comment (shekohex) [inline pallets/mt/src/tests.rs:51]: Try this:
```suggestion
			crate::Error::<Test, _>::ExceedsMaxDefaultHashes,
```

File: pallets/mt/src/lib.rs (hunk 1 above)
Comment (shekohex) [reply pallets/mt/src/lib.rs:313]: Well, the default hashes len is equal to the MaxTreeDepth.
So, I don't see any need for `MaxDefaultHashes` at all.

Comment (Wizdave97) [reply pallets/mt/src/lib.rs:313]: Okay, I'll make that fix

**Pull request merged**